### ISSUE TITLE
Error right after successfully finished obfuscation

### ISSUE THREAD

**Issue by zivkasper** [dashingsoft/pyarmor#616]
**Title**:

Error right after successfully finished obfuscation

**Description**:

Building our app using docker with pyarmor
we got the logging.info('Obfuscate %d scripts OK.', len(files)) log and then the 
raise RuntimeError('Run command failed') error. i looked at the code and can't figure out how that's possible.

our docker code censored :
```
ARG BUILD_PACKAGES="libssl-dev zlib1g-dev wget xz-utils build-essential unzip git"
ARG RUNTIME_PACKAGES="openssl libtool nano nmap tcpdump sudo"


############## The base image, with apt update & upgrade, runtime packages, and all required environment variables
FROM python:3.7.5-slim-buster as build
ARG RUNTIME_PACKAGES
ARG BUILD_PACKAGES


# Install runtime deps
RUN apt update && \
    apt upgrade -y && \
    apt install ${RUNTIME_PACKAGES} -y && \
    apt install ${BUILD_PACKAGES} -y

# We work from the /compile dir to compile all files
WORKDIR /compile


# Install python deps
COPY requirements.txt build_requirements.txt ./

RUN pip install -r requirements.txt && \
    pip install -r build_requirements.txt && \
    git clone https://github.com/dmb2168/cryptography.git && \
	pip install -e cryptography


# Compile python
COPY . .
RUN sh build.sh $COMPILE_ARGS
...
```
build.sh runs a python script which call pyarmor with this args:
```
pyarmor pack --name main -e  -F -y --hidden-import neo4j.packstream.packer --hidden-import neo4j.packstream.unpacker --hidden-import neo4j.bolt.io --hidden-import pkg_resources.py2_warn --hidden-import cmath -x  --exclude=venv,POC,notused --obf-code 2  --clean main.py
```
results
```
INFO     Create pyarmor home path: /root/.pyarmor
INFO     Create trial license file: /root/.pyarmor/license.lic
INFO     Generating public capsule ...
INFO     PyArmor Trial Version 7.0.1
INFO     Python 3.7.5
INFO     Start to register keyfile: pyarmor-regfile-1.zip
INFO     Save registration data to: /root/.pyarmor
INFO     Extracting license.lic
INFO     Extracting .pyarmor_capsule.zip
INFO     This keyfile has been registered successfully.
INFO     PyArmor Version 7.0.1
INFO     Python 3.7.5
INFO     Prepare to pack obfuscated scripts with PyInstaller...
INFO     entry script: main.py
INFO     src for searching scripts: .
INFO     build path: dist/obf
INFO     Run PyArmor to obfuscate scripts...
INFO     
/usr/local/bin/python /usr/local/lib/python3.7/site-packages/pyarmor/pyarmor.py obfuscate -O dist/obf --package-runtime 0 -r --exclude dist --exclude=venv,POC,notused --obf-code 2 ./main.py
INFO     ==================== Run command ====================
INFO     PyArmor Version 7.0.1
INFO     Python 3.7.5
INFO     Target platforms: Native
INFO     Source path is "/compile"
INFO     Entry scripts are ['./main.py']
INFO     Use cached capsule /root/.pyarmor/.pyarmor_capsule.zip
INFO     Search scripts mode: Recursive
INFO     Exclude path "dist"
INFO     Exclude path "venv"
INFO     Exclude path "POC"
INFO     Exclude path "notused"
INFO     Auto exclude output path "dist/obf"
no previously-included directories found matching 'dist'
no previously-included directories found matching 'venv'
no previously-included directories found matching 'dist/obf'
INFO     Save obfuscated scripts to "dist/obf"
INFO     Read product key from capsule
INFO     Obfuscate module mode is 2
INFO     Obfuscate code mode is 2
INFO     Wrap mode is 1
INFO     Restrict mode is 1
INFO     Advanced value is 0
INFO     Super mode is False
INFO     Super plus mode is not enabled
INFO     Generating runtime files to dist/obf
INFO     Extract pytransform.key
INFO     Generate default license file
INFO     Update capsule to add default license file
INFO     Copying /usr/local/lib/python3.7/site-packages/pyarmor/platforms/linux/x86_64/_pytransform.so
INFO     Patch library dist/obf/_pytransform.so
INFO     Patch library file OK
INFO     Copying /usr/local/lib/python3.7/site-packages/pyarmor/pytransform.py
INFO     Generate runtime files OK
INFO     Start obfuscating the scripts...
...
INFO     Obfuscate 1230 scripts OK.
ERROR    Run command failed
```

**Comment by zivkasper** [comment]
ok, so we pinpoint the problem, it's probably something to do with v7.0.1 and caching of dependencies in bitbucket pipelines.

It is solved now, but I would like to request that if subprocess failed can you give return code and maybe error if possible:

I'm referring to this code:
```
def run_command(cmdlist, verbose=True):
    logging.info('\n\n%s\n\n', ' '.join(
        [x if x.find(' ') == -1 else ('"%s"' % x) for x in cmdlist]))
    if verbose:
        sep = '=' * 20
        logging.info('%s Run command %s', sep, sep)
        p = Popen(cmdlist)
        p.wait()
        if p.returncode != 0:
            raise RuntimeError('Run command failed')
        logging.info('%s End command %s\n', sep, sep)
    else:
        p = Popen(cmdlist, stdout=PIPE, stderr=STDOUT)
        output, _ = p.communicate()
        if p.returncode != 0:
            raise RuntimeError(output.decode())
```

**Comment by jondy** [comment]
Thanks, this bug has been fixed in `v7.0.2`.
https://github.com/dashingsoft/pyarmor/releases/tag/v7.0.2
